### ISSUE TITLE
Move from in-tree cloud-controller to external cloud-controller-manager

### ISSUE THREAD

**Issue by fiunchinho** [giantswarm/roadmap#2385]
The in-tree cloud-controller has been deprecated for a while and support will be removed in future releases. We need to start using the external cloud-controller-manager, [like we did in `vintage`](https://github.com/giantswarm/roadmap/issues/888). We already have an app ready https://github.com/giantswarm/aws-cloud-controller-manager-app

The kubelet help says
```--cloud-config (DEPRECATED: will be removed in 1.24 or later, in favor of removing cloud provider code from Kubelet.)```

**Comment by fiunchinho** [comment]
I created a new WC using
- [`cluster-aws` version](https://github.com/giantswarm/cluster-aws/pull/289) that changes the `cloud-provider` parameter from `aws` to `external`
- [`default-apps-aws` version](https://github.com/giantswarm/default-apps-aws/pull/221) that deploys the `aws-cloud-controller-manager`

The cluster came up without issues. 

Then I wanted to test if we were actually using the external cloud-controller. I deleted the `aws-cloud-controller-manager` and added a new worker node. The node has the taint `Taints:             node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule` because the `cloud-controller-manager` is not running.  So I'd say everything is fine.
